File: src/lib/types/transaction.ts
import { z } from 'zod';

export const TransactionType = {
  INCOME: 'income',
  EXPENSE: 'expense',
  TRANSFER: 'transfer'
} as const;

export const transactionSchema = z.object({
  id: z.string().uuid(),
  type: z.enum([TransactionType.INCOME, TransactionType.EXPENSE, TransactionType.TRANSFER]),
  amount: z.number().positive(),
  currency: z.string(),
  category: z.string(),
  subcategory: z.string().optional(),
  date: z.date(),
  description: z.string(),
  paymentMethod: z.string(),
  attachments: z.array(z.string()).optional(),
  tags: z.array(z.string()).optional(),
  notes: z.string().optional(),
  createdAt: z.date(),
  updatedAt: z.date()
});

export type Transaction = z.infer<typeof transactionSchema>;

File: backend/models/transaction.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class TransactionBase(BaseModel):
    type: str
    amount: float
    currency: str
    category: str
    subcategory: Optional[str] = None
    date: datetime
    description: str
    payment_method: str
    notes: Optional[str] = None

class Transaction(TransactionBase):
    id: str
    user_id: str
    created_at: datetime
    updated_at: datetime